docker/docker.go

```diff
@@ -0,0 +1,163 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"net"
+
+	//"os"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/dachad/tcpgoon/debugging"
+)
+
+type IPandPort struct {
+	IP   string
+	Port int
+}
+
+func lookForAvailableLocalPort() (availablePort int){
```

Comment:
up to you, but i would try the same the docker is exposing, and if that fails, require the user to be the one that chooses one